app/aboot/lk2nd-device.c

```diff
@@ -118,13 +134,19 @@ static int lk2nd_find_device_offset(const void *fdt)
 void lk2nd_parse_device_node(const void *fdt)
 {
 	int offset = lk2nd_find_device_offset(fdt);
+	int len = 0;
 	if (offset < 0) {
 		dprintf(CRITICAL, "Failed to find matching lk2nd,device node: %d\n", offset);
 		return;
 	}
 
 	lk2nd_samsung_muic_reset(fdt, offset);
 
+	fdt_getprop(fdt, offset, "lk2nd,motorola-unitinfo", &len);
+	if (len > 0) {
+		lk2nd_motorola_smem_write_unit_info(fdt);
```

Comment:
Can you check the prop inside `lk2nd_motorola_smem_write_unit_info` (just like `lk2nd_samsung_muic_reset` does)?